test/test-thread.c

```diff
@@ -294,3 +298,18 @@ TEST_IMPL(thread_stack_size_explicit) {
 
   return 0;
 }
+
+
+/* Since we can't check the detached state of a thread from the thread itself,
+ * we pass the expected state as an argument to the thread function.
+ */
+static void detach_check(void* arg) {
+  ASSERT_PTR_EQ(arg, &options);
```

Comment:
This check doesn't really prove anything. I'd just remove it, drop `options` (confusing - reads like you're passing config to uv_thread_create) and make `thread` function-local to the test.

Reply:
Makes sense.